components/ai_chat/core/browser/engine/engine_consumer_oai.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_ENGINE_CONSUMER_OAI_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_ENGINE_CONSUMER_OAI_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/browser/engine/oai_api_client.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-forward.h"
+
+namespace api_request_helper {
+class APIRequestResult;
+}  // namespace api_request_helper
+
+namespace network {
+class SharedURLLoaderFactory;
+}  // namespace network
+
+namespace ai_chat {
+
+using api_request_helper::APIRequestResult;
+
+class EngineConsumerOAIRemote : public EngineConsumer {
```

Comment:
This class needs unit tests - especially around how it converts incoming function call params to API messages, as well as parsing of responses such as suggested questions.

Reply:
added a unittest for this class. however, tested only suggested questions query. i'll add others soon.